ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/BonsaiWorldStateArchive.java

```diff
@@ -94,71 +131,71 @@ public Optional<WorldState> get(final Hash rootHash, final Hash blockHash) {
 
   @Override
   public boolean isWorldStateAvailable(final Hash rootHash, final Hash blockHash) {
-    return trieLogManager.getBonsaiLayeredWorldState(blockHash).isPresent()
+    return trieLogManager.getBonsaiCachedWorldState(blockHash).isPresent()
         || persistedState.blockHash().equals(blockHash)
         || worldStateStorage.isWorldStateAvailable(rootHash, blockHash);
   }
 
-  public Optional<SnapshotMutableWorldState> getMutableSnapshot(final Hash blockHash) {
+  public Optional<MutableWorldState> getMutableSnapshot(final Hash blockHash) {
     return rollMutableStateToBlockHash(
             BonsaiSnapshotWorldState.create(this, worldStateStorage), blockHash)
         .map(SnapshotMutableWorldState.class::cast);
   }
 
   @Override
   public Optional<MutableWorldState> getMutable(
-      final long blockNumber, final boolean isPersistingState) {
-    final Optional<Hash> blockHashByNumber = blockchain.getBlockHashByNumber(blockNumber);
-    if (blockHashByNumber.isPresent()) {
-      return getMutable(null, blockHashByNumber.get(), isPersistingState);
+      final Hash rootHash, final Hash blockHash, final boolean shouldPersistState) {
```

Comment:
rootHash is only needed if shouldPersistState == true, and in that case it is not passed any further along. Could be deleted.